### ISSUE TITLE
Discussion about the section 5.2 of the paper

### ISSUE THREAD

**Issue by YouJiacheng** [MineDojo/MineDojo#25]
Dear MineDojo Team:
I am following your brilliant MineDojo Project. Both knowledge base and proposed MineCLIP method in this project are great and impressive.
But I have a minor concern about the section 5.2 of the paper:
![image](https://user-images.githubusercontent.com/83971976/182381322-1ab38279-75c7-467e-a782-54b0c410b55b.png)
It was said that VideoCLIP use a monolithic model S3D as $\phi_V$. However, AFAICT, this is a misunderstanding caused by the misleading terminology used in VideoCLIP.
![image](https://user-images.githubusercontent.com/83971976/182381739-23e2303b-4172-4a9a-abbc-bd5fbade93d9.png)
In the section 5.3 of VideoCLIP, it was said that S3D serves as the "Video Encoder" in VideoCLIP. However, S3D in VideoCLIP doesn't encode the whole video clip(a few seconds). It merely encode 30 frames in 1 second (30fps video used) into 1 video token, which will be aggregated with other video tokens in the clip by transformer.

See also https://github.com/facebookresearch/fairseq/tree/main/examples/MMPT#demo-of-inference, S3D won't perform interaction along `T` axis.
```python
# B, T, FPS, H, W, C (VideoCLIP is trained on 30 fps of s3d)
video_frames = torch.randn(1, 2, 30, 224, 224, 3)
caps, cmasks = aligner._build_text_seq(
    tokenizer("some text", add_special_tokens=False)["input_ids"]
)

caps, cmasks = caps[None, :], cmasks[None, :]  # bsz=1

with torch.no_grad():
    output = model(video_frames, caps, cmasks, return_score=True)
print(output["score"])  # dot-product
```

In another word, the "Video Encoder" in VideoCLIP is not $\phi_V$. It actually is a "Sub-Video"/"Super-Frame"/"Super-Image" Encoder. Thus, the "Sub-Video" or "Super-Frame" features can be cached just like image features in MineCLIP, and VideoCLIP's $phi_V$ is factorized as well.

In addition, S3D in VideoCLIP is completely frozen, and thus its "Sub-Video" features can be pre-computed and 30x input size might be acceptable.

**Comment by yunfanjiang** [comment]
Hey @YouJiacheng , thank you for initiating this discussion! 

Yes, you are absolutely correct about VideoCLIP and S3D. They first encode every 30 consecutive frames to a single S3D feature, then concatenate with other video tokens and finally aggregate.

Things become different in MineCLIP. In the second point of the paragraph you took screenshot, we mentioned that the pre-trained MineCLIP also provides a very good RGB image encoder for the agent. Note that our agent takes a single RGB frame every timestep. Our MineCLIP enables 1) extracting single-image feature for the agent and 2) calculating video features for reward calculation at the same time with the feature caching mechanism. This would not be possible for VideoCLIP with S3D, which doesn't preserve single-frame image features.

Hope you find this elaboration helpful. Feel free to let me know if you have any other questions.

**Comment by yunfanjiang** [comment]
Closing this issue now. Feel free to reopen if you have further questions to discuss. Thanks for your interest again.

**Comment by YouJiacheng** [comment]
Thanks for your reply!

Yes, MineCLIP is designed for cooperating with the agent, which uses 1 frame every timestep. This brilliant design combines the agent and the learned reward seamlessly, enables sharing 1-step features calculation between them and makes the learned reward nearly free. VideoCLIP cannot shares computation with the agent, unless the agent also takes 30 frames every timestep, which is not common in vision-based RL.

Would MineDojo team amend the paragraph I took screenshot in future version of the paper? This paragraph actually misled me until I dived deeper in VideoCLIP.
